scipy/linalg/tests/test_special_matrices.py

```diff
@@ -640,3 +642,45 @@ def test_fiedler_companion():
     fc = fiedler_companion([1., -16., 86., -176., 105.])
     assert_array_almost_equal(eigvals(fc),
                               np.array([7., 5., 3., 1.]))
+
+
+def test_convolution_matrix():
+    'test convolution_matrix vs. numpy.convolve for various parameters'
+
+    def test_vector(n, cpx):
+        'make a complex or real test vector of length n'
+        if cpx:
+            return np.random.normal(size=(n, 2))\
+                .astype(np.float64).view(np.complex128).ravel()
+        else:
+            return np.random.normal(size=(n,))
+
+    # first arg must be a 1d array, otherwise ValueError
+    with assert_raises(ValueError):
+        convolution_matrix(1, 4)
+
+    # mode must be in ('full','valid','same')
+    with assert_raises(ValueError):
+        convolution_matrix((1, 1), 4, mode='invalid argument')
+
+    array_sizes = (2, 9, 100)
+    for cpx, na, nv, mode in itertools.product(
```

Comment:
I suppose this is still used in our codebase a fair bit; at some number of permutations I think we should prefer to parametrize, but anyway..